Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -0,0 +1,478 @@
+                    ++size;
+                }
+            }
+        }
+        return size;
+    }
+
+    void Path::FormatString(char* stringBuffer, size_t bufferSize) const
+    {
+        size_t bufferIndex = 0;
+
+        auto putChar = [&](char c)
+        {
+            if (bufferIndex == bufferSize)
+            {
+                return;
+            }
+            stringBuffer[bufferIndex++] = c;
+        };
+
+        auto writeToBuffer = [&](const char* key)
+        {
+            for (size_t keyIndex = 0; key[keyIndex]; ++keyIndex)
+            {
+                const char c = key[keyIndex];
+                if (c == EscapeCharacter)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(TildeSequence);
+                }
+                else if (c == PathSeparator)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(ForwardSlashSequence);
+                }
+                else
+                {
+                    putChar(c);
+                }
+            }
+        };
+
+        for (const PathEntry& entry : m_entries)
+        {
+            putChar(PathSeparator);
+            if (entry.IsEndOfArray())
+            {
+                putChar(EndOfArrayCharacter);
+            }
+            else if (entry.IsIndex())
+            {
+                bufferIndex += azsnprintf(&stringBuffer[bufferIndex], bufferSize - bufferIndex, "%zu", entry.GetIndex());
+            }
+            else
+            {
+                writeToBuffer(entry.GetKey().GetCStr());
+            }
+        }
+
+        putChar('\0');
+    }
+
+    AZStd::string Path::ToString() const
+    {
+        AZStd::string formattedString;
+        const size_t size = GetStringLength();
+        formattedString.resize_no_construct(size);
+        FormatString(formattedString.data(), size + 1);
+        return formattedString;
+    }
+
+    void Path::AppendToString(AZStd::string& output) const
+    {
+        const size_t startIndex = output.length();
+        const size_t stringLength = GetStringLength();
+        output.resize_no_construct(startIndex + stringLength);
+        FormatString(output.data() + startIndex, stringLength + 1);
+    }
+
+    void Path::FromString(AZStd::string_view pathString)
+    {
+        m_entries.clear();
+        if (pathString.empty())
+        {
+            return;
+        }
+
+        size_t pathEntryCount = 0;
+        for (size_t i = 1; i <= pathString.size(); ++i)
+        {
+            if (pathString[i] == PathSeparator)
+            {
+                ++pathEntryCount;
+            }
+        }
+        m_entries.reserve(pathEntryCount);
+
+        // Ignore a preceeding path separator and start processing after it
+        size_t pathIndex = pathString[0] == PathSeparator ? 1 : 0;
+        bool isNumber = true;
+        AZStd::string convertedSection;
+        for (size_t i = pathIndex; i <= pathString.size(); ++i)
```

Comment:
Actually instead of using a for loop you could just call [AZ::StringFunc::TokenizeVisitor](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/StringFunc/StringFunc.h#L273-L292) and supply a lambda 

```
auto ParseSegment( = [this]AZStd::string_view entry)
{
   // Simplified block logic below
};
// Skip initial path separator
AZ::StringFunc::TokenizeVisitor(pathString.substr(1);, ParseSegment, PathSeparator, true, true);
```